agent/lib_guzzle6.c

```diff
@@ -458,33 +408,142 @@ void nr_guzzle6_enable(TSRMLS_D) {
       "      return $promise;"
       "    };"
       "  }"
-      "}",
-      NULL, "newrelic/Guzzle6" TSRMLS_CC);
+      "}", guzzle_version, guzzle_version, guzzle_version);
+
+  char *guzzle_ver = nr_formatf("newrelic/Guzzle%d", guzzle_version);
+  _retval = zend_eval_string(eval, NULL, guzzle_ver TSRMLS_CC);
 
-  if (SUCCESS == retval) {
+  if (SUCCESS == _retval && guzzle_version == 6) {
     nr_php_wrap_user_function(NR_PSTR("GuzzleHttp\\Client::__construct"),
-                              nr_guzzle_client_construct TSRMLS_CC);
-  } else {
+                              nr_guzzle6_client_construct TSRMLS_CC);
+  }else if (SUCCESS == _retval && guzzle_version == 7){
+    nr_php_wrap_user_function(NR_PSTR("GuzzleHttp\\Client::__construct"),
+                              nr_guzzle7_client_construct TSRMLS_CC);
+  }else {
     nrl_warning(NRL_FRAMEWORK,
                 "%s: error evaluating PHP code; not installing handler",
                 __func__);
   }
 }
 
-void nr_guzzle6_minit(TSRMLS_D) {
-  zend_class_entry ce;
+NR_PHP_WRAPPER_START(nr_guzzle_client_construct_helper){
+  zval* this_var = nr_php_scope_get(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  zval* retval;
+  int guzzle_version = 6;
+  char *version = nr_guzzle_version(this_var);
+  if (php_version_compare(version, "7") >= 0){
+    guzzle_version = 7;
+  }
 
-  if (0 == NRINI(guzzle_enabled)) {
-    return;
+  (void)wraprec;
+  NR_UNUSED_SPECIALFN;
+  NR_PHP_WRAPPER_CALL;
+  /*
+   * Get our middleware callable (which is just a string), and make sure it's
+   * actually callable before we invoke push(). (See also PHP-1184.)
+   */
+  char *str_middleware = nr_formatf("newrelic\\Guzzle%d\\middleware", 
```

Comment:
I don't see this being freed.

Reply:
Thank you for pointing that out! Fixed [29f4551](https://github.com/newrelic/newrelic-php-agent/pull/498/commits/29f45519674eb0cf75e552d5e1e06bf478342776)